### ISSUE TITLE
cast estimate --create requests password

### ISSUE THREAD

**Issue by gbalabasquer** [foundry-rs/foundry#4035]
### Component

Cast

### Have you ensured that all of these are up to date?

- [X] Foundry
- [ ] Foundryup

### What version of Foundry are you on?

cast 0.2.0 (a44159a 2023-01-02T00:04:45.597843Z)

### What command(s) is the bug in?

cast estimate

### Operating System

macOS (Apple Silicon)

### Describe the bug

`cast estimate --create "$BYTECODE" ...` shouldn't prompt password request.

**Comment by hexonaut** [comment]
Related comment: https://github.com/foundry-rs/foundry/pull/4038#issuecomment-1379443024

**Comment by gbalabasquer** [comment]
@mds1 @mattsse I've noticed this issue is happening again in the latest versions.

**Comment by mds1** [comment]
Do you have a command to reproduce this, including which env vars like ETH_FROM or ETH_GAS_PRICE are set? I just tried on the latest with the below command and it doesn't seem to be prompting me for a password:

```
$ ETH_FROM=0x209884F676AF1b993dE5d453Ee38eFD0082D6cCa ETH_GAS_PRICE=1 cast estimate --create 0x
53000
```

**Comment by gbalabasquer** [comment]
Yeah that example works the same for me, but I think I found what the problem is, if the address that I set as `ETH_FROM` is in the path defined by `ETH_KEYSTORE` it will request for the password. Is that an expected behavior?
